msgraphcore/client_factory.py

```diff
@@ -0,0 +1,64 @@
+import functools
+from typing import Optional
+
+from requests import Session
+
+from msgraphcore.constants import CONNECTION_TIMEOUT, REQUEST_TIMEOUT
+from msgraphcore.enums import APIVersion, NationalClouds
+from msgraphcore.middleware.abc_token_credential import TokenCredential
+from msgraphcore.middleware.authorization import AuthorizationHandler
+from msgraphcore.middleware.middleware import BaseMiddleware, MiddlewarePipeline
+
+
+class HTTPClientFactory:
+    """
+    Constructs HTTP Client(session) instances configured with either custom or default
+    pipeline of middleware.
+    """
+    def __init__(self, **kwargs):
```

Comment:
It will be a better experience for devs to get intellisense for the expected parameters. Using **kwargs as is means devs don't get intellisense.